File: simple-stack-example-multistack/src/main/java/com/zhuinden/simplestackdemomultistack/presentation/paths/main/cloudsync/another/AnotherKey.java
package com.zhuinden.simplestackdemomultistack.presentation.paths.main.cloudsync.another;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestack.Services;
import com.zhuinden.simplestackdemomultistack.R;
import com.zhuinden.simplestackdemomultistack.application.Key;
import com.zhuinden.simplestackdemomultistack.application.MainActivity;

/**
 * Created by Zhuinden on 2017.02.19..
 */
@AutoValue
public abstract class AnotherKey
        extends Key
        implements Services.Child {
    @Override
    public int layout() {
        return R.layout.path_another;
    }

    @Override
    public String stackIdentifier() {
        return MainActivity.StackType.CLOUDSYNC.name();
    }

    public static Object create(Key parent) {
        return new AutoValue_AnotherKey(parent);
    }
}

File: simple-stack-example-multistack-view/src/main/java/com/zhuinden/simplestackdemomultistack/features/main/cloudsync/another/AnotherKey.kt
package com.zhuinden.simplestackdemomultistack.features.main.cloudsync.another

import com.zhuinden.simplestackdemomultistack.R
import com.zhuinden.simplestackdemomultistack.application.MainActivity
import com.zhuinden.simplestackdemomultistack.core.navigation.MultistackKey
import kotlinx.android.parcel.Parcelize

@Parcelize
data class AnotherKey(private val placeholder: String = "") : MultistackKey() {
    override fun layout(): Int = R.layout.another_view
    override fun stackIdentifier(): String = MainActivity.StackType.CLOUDSYNC.name
}
